lib/logflare/google/resource_manager.ex

```diff
@@ -12,85 +12,75 @@ defmodule Logflare.Google.CloudResourceManager do
   alias Logflare.TeamUsers
   alias Logflare.Billing
 
-  defp env_project_number, do: Application.get_env(:logflare, Logflare.Google)[:project_number]
-  defp env_service_account, do: Application.get_env(:logflare, Logflare.Google)[:service_account]
-  defp env_api_sa, do: Application.get_env(:logflare, Logflare.Google)[:api_sa]
-
-  defp env_compute_engine_sa,
-    do: Application.get_env(:logflare, Logflare.Google)[:compute_engine_sa]
-
-  defp env_cloud_build_sa, do: Application.get_env(:logflare, Logflare.Google)[:cloud_build_sa]
+  def list_projects() do
+    conn = GenUtils.get_conn()
+    Api.Projects.cloudresourcemanager_projects_list(conn)
+  end
 
   def get_iam_policy() do
     conn = GenUtils.get_conn()
 
     body = %Model.GetIamPolicyRequest{}
 
-    Api.Projects.cloudresourcemanager_projects_get_iam_policy(conn, env_project_number(),
+    Api.Projects.cloudresourcemanager_projects_get_iam_policy(
+      conn,
+      env_project_number(),
       body: body
     )
   end
 
-  def set_iam_policy() do
-    conn = GenUtils.get_conn()
+  def set_iam_policy(opts \\ [async: true])
 
-    Task.Supervisor.start_child(Logflare.TaskSupervisor, fn ->
-      members = build_members()
-
-      bindings =
-        [
-          %Model.Binding{
-            members: members,
-            role: "roles/bigquery.jobUser"
-          }
-        ] ++ get_service_accounts()
-
-      policy = %Model.Policy{
-        bindings: bindings
-      }
-
-      body = %Model.SetIamPolicyRequest{
-        policy: policy
-      }
+  def set_iam_policy(async: true) do
+    Task.Supervisor.start_child(Logflare.TaskSupervisor, fn -> set_iam_policy(async: false) end)
+  end
 
-      {f, a} = __ENV__.function
-      fun = "#{f}" <> "_" <> "#{a}"
-
-      case Api.Projects.cloudresourcemanager_projects_set_iam_policy(conn, env_project_number(),
-             body: body
-           ) do
-        {:ok, _response} ->
-          Logger.info("Set IAM policy: #{Enum.count(members)} accounts",
-            logflare: %{
-              google: %{
-                cloudresourcemanager: %{
-                  "#{fun}": %{accounts: Enum.count(members), response: :ok}
+  def set_iam_policy(async: false) do
+    conn = GenUtils.get_conn()
+    members = build_members()
+
+    bindings =
+      [%Model.Binding{members: members, role: "roles/bigquery.jobUser"}] ++
+        get_service_accounts()
+
+    policy = %Model.Policy{bindings: bindings}
+    body = %Model.SetIamPolicyRequest{policy: policy}
+
+    {function, arity} = __ENV__.function
+    caller = "#{function}" <> "_" <> "#{arity}"
+
+    case Api.Projects.cloudresourcemanager_projects_set_iam_policy(conn, env_project_number(),
```

Comment:
it would be simpler to mock this function and assert on the received arguments in the mock

For example:
```
# in test
Api.Projects
|> expect(:cloudresourcemanager_projects_set_iam_policy, fn conn, proj_num-> 
  assert [%_{...} | _ ] = body.policy.bindings 
  {:ok, %{}}
end)
```
likewise for all the other `Api.Projects.*` apis.
Mocking the dep directly makes it easier to check that:
1. args are passed correctly
2. 3rd party func is called correctly
3. Tesla's not-so-nice mocking interface is not used

Reply:
Plus the argument that this is testing the Google Library... damn it yeah this is way more correct with mimic... will change 👍

Reply:
I had to do the assertions on the stub otherwise it would call Tesla 🤷